### ISSUE TITLE
(FunctionClauseError) no function clause matching in IO.chardata_to_string/1

### ISSUE THREAD

**Issue by duane-strikwerda** [parroty/excoveralls#273]
Hello:

Phoenix      1.5.13
erlang         24.1.7 [erts-12.1.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]
elixir           1.13.0-otp-24 (compiled with Erlang/OTP 24)
excoveralls 0.14.0

(Windows running WSL 2)

> MIX_ENV=test mix coveralls.html

![excoveralls error](https://user-images.githubusercontent.com/69985640/145640234-be780c72-b27c-4aff-b432-3ddb5abfb486.PNG)

ExCoveralls was working just fine on elixir 1.12.2 - but seems to have stopped, I am unsure if it is the upgrade, or ?

Any thoughts/hints would be a great help

Thanks
-d

**Comment by MrGrinst** [comment]
Seeing this same issue on 1.13.0.

```
Excluding tags: [:integration_test]
.....................................
Finished in 0.6 seconds (0.6s async, 0.00s sync)
7 doctests, 30 tests, 0 failures
Randomized with seed 375316
Wrote JUnit report to: /app/_build/test/lib/shared_db-junit.xml
** (FunctionClauseError) no function clause matching in IO.chardata_to_string/1    
    
    The following arguments were given to IO.chardata_to_string/1:
    
        # 1
        nil
    
    Attempted function clauses (showing 2 out of 2):
    
        def chardata_to_string(+string+) when -is_binary(string)-
        def chardata_to_string(+list+) when -is_list(list)-
    
    (elixir 1.13.0) lib/io.ex:627: IO.chardata_to_string/1
    (elixir 1.13.0) lib/path.ex:423: Path.dirname/1
    lib/excoveralls/cover.ex:21: ExCoveralls.Cover.module_path/1
    lib/excoveralls/stats.ex:54: ExCoveralls.Stats.add_counts/4
    (elixir 1.13.0) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/excoveralls/stats.ex:28: ExCoveralls.Stats.report/1
    lib/excoveralls.ex:45: ExCoveralls.execute/2
    (mix 1.13.0) lib/mix/tasks/test.ex:538: Mix.Tasks.Test.do_run/3
    (mix 1.13.0) lib/mix/task.ex:397: anonymous fn/3 in Mix.Task.run_task/3
    (mix 1.13.0) lib/mix/task.ex:451: Mix.Task.run_alias/5
    (mix 1.13.0) lib/mix/project.ex:396: Mix.Project.in_project/4
    (elixir 1.13.0) lib/file.ex:1560: File.cd!/2
    (mix 1.13.0) lib/mix/task.ex:531: anonymous fn/4 in Mix.Task.recur/1
    (elixir 1.13.0) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
    (mix 1.13.0) lib/mix/task.ex:530: Mix.Task.recur/1
    (mix 1.13.0) lib/mix/project_stack.ex:221: Mix.ProjectStack.recur/1
    (mix 1.13.0) lib/mix/task.ex:451: Mix.Task.run_alias/5
    lib/mix/tasks.ex:54: Mix.Tasks.Coveralls.do_run/2
    (mix 1.13.0) lib/mix/task.ex:397: anonymous fn/3 in Mix.Task.run_task/3
    (mix 1.13.0) lib/mix/task.ex:455: Mix.Task.run_alias/5
```

**Comment by christhekeele** [comment]
I'm getting this as well, on all versions of `1.13`, no matter the otp I use!

**Comment by ryanzidago** [comment]
Can confirm as well. As soon as I updated to Elixir `1.13.0` and Erlang `24.0.1`, I saw the above mentioned error.

**Comment by sudoku-lord** [comment]
Yup, I third this. Had the same issue with Erlang 24.0 and Elixir 1.13 (on the command line, at least). Bumped down to Erlang 22.0 and 1.12 and it ran just fine in my GitHub Action.

**Comment by parroty** [comment]
Thank you for the report. I'm struggling to reproduce the error, but I'm wondering if the following change might be related.
The error still occurs with `excoveralls v0.14.4`? (If it's older than this version, is it possible to try with v0.14.4?).

- https://github.com/parroty/excoveralls/pull/267

**Comment by christhekeele** [comment]
I just walked through the latest versions—it does indeed work on `v0.14.4` but nothing earlier in the `v0.14` line.

**Comment by duane-strikwerda** [comment]
`mix deps.update excoveralls`

resulted in:

```
Upgraded:
  excoveralls 0.14.2 => 0.14.4
  jason 1.2.2 => 1.3.0
* Updating jason (Hex package)
* Updating excoveralls (Hex package)
```

which seems to have fixed the issue!

Many thanks again!

**Comment by ks1venberg** [comment]
> Thank you for the report. I'm struggling to reproduce the error, but I'm wondering if the following change might be related. The error still occurs with `excoveralls v0.14.4`? (If it's older than this version, is it possible to try with v0.14.4?).
> 
> * [Support Elixir 1.13 #267](https://github.com/parroty/excoveralls/pull/267)

Thank you! The problem was solved by installing v 0.14.4
In mix.exs I've assigned: {:excoveralls, ">= 0.14.0"}

**Comment by christhekeele** [comment]
@ks1venberg do be careful with with `{:excoveralls, ">= 0.14.0"}`, as that will eagerly install any update, including major breaking versions. I'd recommending pinning it to something with `~>` as described in [the version docs](https://hexdocs.pm/elixir/Version.html#module-requirements).
